data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/S3ObjectWorkerIT.java

```diff
@@ -110,13 +109,22 @@ private S3ObjectWorker createObjectUnderTest(final Codec codec, final int number
     void parseS3Object_correctly_loads_data_into_Buffer(
             final RecordsGenerator recordsGenerator,
             final int numberOfRecords,
-            final int numberOfRecordsToAccumulate) throws Exception {
+            final int numberOfRecordsToAccumulate,
+            final boolean shouldCompress) throws Exception {
 
-        final String key = "s3source/s3/" + numberOfRecords + "_" + Instant.now().toString() + recordsGenerator.getFileExtension();
-        s3ObjectGenerator.write(numberOfRecords, key, recordsGenerator);
+        S3ObjectWorker objectUnderTest;
+        String key;
 
-        final S3ObjectWorker objectUnderTest = createObjectUnderTest(recordsGenerator.getCodec(), numberOfRecordsToAccumulate);
+        if (shouldCompress) {
+            key = "s3source/s3/" + numberOfRecords + "_" + Instant.now().toString() + recordsGenerator.getFileExtension() + ".gz";
```

Comment:
This string is duplicated. I'd suggest refactoring it so that only the `.gz` is added to the end afterward.

You could make a private function that uses a non-final `key`. Then when compression is enabled, it adds the `.gz` extension.